libraries/AP_Scripting/applets/Script_Controller.lua

```diff
@@ -157,6 +162,55 @@ function copy_scripts(subdir)
    return ret
 end
 
+ --[[ 
+   load a mission from a MISSION_FILENAME file in subdirectory n
+ --]]
+function mission_load(n)
+  file_name = get_scripts_dir() .. "/" .. n .."/" .. MISSION_FILENAME
+  -- Open file
+  file = io.open(file_name)
+  if not file then
+     return
+  end
+  -- check header
+  assert(string.find(file:read('l'),'QGC WPL 110') == 1, file_name .. ': incorrect format')
+  -- clear any existing mission
+  assert(mission:clear(), 'Could not clear current mission')
+  -- read each line and write to mission
+  local item = mavlink_mission_item_int_t()
+  local index = 0
+  while true do
+    local data = {}
+    for i = 1, 12 do
+      data[i] = file:read('n')
```

Comment:
this is quite a fragile way of doing parsing. If the mission file has one less column than expected then it will still load, but will end up with a corrupted mission
using file:read() with no argument will get one line, then use string.find() to parse that line will be much more robust.
See load_trick() in the main aerobatics script for an example
It would also be good to check the sequence number when parsing, and if not the expected seq number then fail the whole mission load.

Reply:
done